src/sagemaker/dataset_definition/inputs.py

```diff
@@ -141,3 +215,22 @@ class S3Input(ApiObject):
     s3_input_mode = "File"
     s3_data_distribution_type = "FullyReplicated"
     s3_compression_type = None
+
```

Comment:
It is not necessary to have both the instance and class variables here. One or the other is enough. I suggest deleting the class variables. Mufi and I discussed this on a call.